### DISCUSSION TITLE
Is there a heartbeat check interface

### DISCUSSION THREAD

**Discussion by mitht** [open-telemetry/opentelemetry-java-instrumentation#11049] [Q&A] [answered]
How to check if the OpenTelemetry agent is still running through heartbeat detection? Does OpenTelemetry provide a probing interface?

**Comment by jack-berg** [comment]
Transferring to opentelemetry-java-instrumentation, since this repo doesn't contain the agent.

**Comment by jack-berg** [comment]
The otel java agent doesn't run as a separate process, so a traditional heartbeat isn't really applicable. If you haven't disabled the [runtime-telemetry](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/runtime-telemetry) instrumentation, you should be able to reliably see runtime telemetry metrics exported on a regular interval. This can be used as a sort of heartbeat.

**Reply by mitht** [reply]
Which metric do you recommend to use?

**Reply by trask** [reply, marked as answer]
any of them should be ok to use as a heartbeat

also check out https://github.com/open-telemetry/opentelemetry-specification/issues/2923
